lib/magma/actions/add_model.rb

```diff
@@ -0,0 +1,114 @@
+class Magma
+  class AddModelAction < BaseAction
+    def perform
+      @model = create_model
+
+      if !table_parent_link?
+        @model.identifier(@action_params[:identifier].to_sym).save
+      end
+
+      @model.parent(@action_params[:parent_model_name].to_sym).save
+
+      project.models[@model.model_name] = @model
+
+      parent_model.
+        send(@action_params[:parent_link_type], @model.model_name).
+        save
+
+      true
+    end
+
+    def rollback
+      parent_model.attributes.delete(@model.model_name)
+      project.unload_model(@model.model_name)
+    end
+
+    private
+
+    def create_model
+      Magma.instance.db[:models].insert(
+        project_name: @project_name,
+        model_name: @action_params[:model_name]
+      )
+
+      project.load_model(
+        project_name: @project_name,
+        model_name: @action_params[:model_name]
+      )
+    end
+
+    def validations
+      [
+        :validate_required_fields,
+        :validate_parent_model,
+        :validate_parent_link_type
+      ]
+    end
+
+    def validate_required_fields
+      required_fields.each do |field|
+        validate_pressence(field)
```

Comment:
This is the only issue i can find in this PR.

Reply:
Spelling is hard 🙈. Fixed in https://github.com/mountetna/magma/pull/182/commits/dc23c64f66560027845a902c490e0eefe899c634.